[10.0][FIX] Wrong outstanding amounts for partial reconciliations

MiquelRForgeFlow opened the pull request:
Fixes https://github.com/OCA/account-financial-reporting/issues/546. Backport of https://github.com/OCA/account-financial-reporting/pull/547 and https://github.com/OCA/account-financial-reporting/pull/565.

MiquelRForgeFlow committed:
[FIX] Wrong outstanding amounts for partial reconciliations

gdgellatly reviewed (approved):
Tested corresponding patch for V12 against live data and confirm working. Code comparison here with that patch. No test.

gdgellatly committed:
[FIX] Aging Buckets

In certain scenarios involving mutliple payments for multiple
invoices, where the reconciliation happens together and the debit
move is the smaller item the aging duplicates its value, making
the balance greater than it should be.

This fix works by preventing moves where different lines are both partial
reconciled being grouped together by adding the lines id
to the group by clause of _show_buckets_sql_q1

dominiquekonsuntack commented:
in v10,
the PDF report also shows the invoices that are paid (balance = 0.00)
